g2p/app.py

```diff
@@ -123,30 +145,26 @@ def return_echart_data(tg: Union[CompositeTransductionGraph, TransductionGraph])
     return nodes, edges
 
 
-@APP.route("/")
-def home():
+@APP.get("/", response_class=HTMLResponse)
+def home(request: Request):
     """Return homepage of g2p studio"""
-    return render_template("index.html")
+    return TEMPLATES.TemplateResponse("index.html", {"request": request})
 
 
-@APP.route("/docs")
-def docs():
-    """Return swagger docs of g2p studio API"""
-    return render_template("docs.html")
-
-
-@SOCKETIO.on("conversion event", namespace="/convert")
-def convert(message):
+@APP.sio.on("conversion event", namespace="/convert")  # type: ignore
+async def convert(sid, message):
```

Comment:
That's weird, Codecov seems to think this function is not at all covered in CI testing. I find that hard to believe since it's the main function in the API! `test_studio.py` should trigger calls to this, right?

Reply:
yes something is weird with codecov, I'll need to fix that before merging

Reply:
Actually, not sure it's codecov: I ran it locally with `coverage run` and got the same answer, which I still don't understand.

Reply:
It is probably an artifact of the switch to uvicorn and asyncio, which, apparently, doesn't really do signal handlers well.  Killing with SIGINT and waiting a bit longer *seems* to fix it.